### ISSUE TITLE
ECUtil.getNamedCurveOid does not safely handle strings with space at last position

### ISSUE THREAD

**Issue by kares** [bcgit/bc-java#1869]
e.g. 
```java
org.bouncycastle.jcajce.provider.asymmetric.util.ECUtil.getNamedCurveOid(" ");
```
is going to lead to a `java.lang.StringIndexOutOfBoundsException: String index out of range: 0` [due the assumption](https://github.com/bcgit/bc-java/blob/1.78.1/prov/src/main/java/org/bouncycastle/jcajce/provider/asymmetric/util/ECUtil.java#L325) that after a `' '` char there's still another character

**Comment by dghgit** [comment]
I've changed it to now do a trim() after the null check.

**Comment by dghgit** [comment]
Fixed in 1.79.
